cmd/endpoint.go

```diff
@@ -349,12 +349,21 @@ func CreateEndpoints(serverAddr string, args ...string) (string, EndpointList, S
 					err = fmt.Errorf("path '%s' can not be served by different port on same address", endpoint.Path)
 					return serverAddr, endpoints, setupType, err
 				}
-
 				pathIPMap[endpoint.Path] = IPSet.Union(hostIPSet)
 			} else {
 				pathIPMap[endpoint.Path] = hostIPSet
 			}
 		}
+		for _, endpoint := range endpoints {
+			if IPSet, ok := pathIPMap[endpoint.Path]; ok {
+				// Check if all IPs in the path IP map are local, we do not allow more than
+				// one IPs on same server to serve same disk paths.
+				if len(localIP4.Intersection(IPSet)) > 1 {
+					err = fmt.Errorf("path '%s' cannot be served by different address on same server", endpoint.Path)
+					return serverAddr, endpoints, setupType, err
```

Comment:
This logic does not work in below case.
* on `node1`, command `minio server http://node1/d1 http://node2/d2 http://node3/d3 http://node4/d4` is run.
* `node1` has three IP addresses, hence `localIP4` is set to `[10.1.10.1 127.0.0.1 192.168.1.1]`.
* `node1`  resolves to two IP addresses `[10.1.10.1 192.168.1.1]`.
* `pathIPMap` contains `{/d1: [10.1.10.1 192.168.1.1], ...}`.
* Now this check errors out.

Right way of doing this is, check `Path` of endpoints are not same where `IsLocal` field of endpoints is set.  Right place to insert this logic is a endpoint.go:373

Reply:
done